paper-api/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java

```diff
@@ -86,6 +92,30 @@ public Player getEntity() {
         return (Player) this.entity;
     }
 
+    /**
+     * Get whether the death message should be shown.
+     * By default, this is determined by {@link org.bukkit.GameRule#SHOW_DEATH_MESSAGES}.
+     *
+     * @return whether the death message should be shown
+     * @see #deathChatMessage()
+     * @see #deathScreenMessage()
+     */
+    public boolean allowDisplayingDeathMessage() {
```

Comment:
I am a bit confused on the usecase for this.

In case the gamerule is true, plugins can already null it out to avoid sending the messages.
In case it is false (the non default) it feels pretty weird for a plugin to change this here?

Reply:
I added this based on a TODO which was in the code and also to expose the gameRule... We already disccussed this with NotTamion above (https://github.com/PaperMC/Paper/pull/12221#discussion_r1983046406) But I can remove it, if you want...

Reply:
Ah, the discussion was marked as resolved, I did not see it :sweat_smile: mb
I guess the magic boolean has always been there, so exposing it is a general good change.
I'll chat with owen on the naming of this, I think I'd prefer this being named *very* closely after the gamerule and/or include the fact that  this is a gamerule in the method name but, that is pending further chats with other team members.

Reply:
So: 

getShowDeathMessagesGameRule / setShowDeathMessagesGameRule would be better?

but then again, you are not actually changing the gamerule here... so maybe with Override suffix?

Reply:
mhm maybe, tho those names are quite verbose. Might be okay to not include the gamerule suffix, I'll think a bit about it throughout the day. Worse case we can spam the javadocs a bit about how this isn't really a boolean added by us but is direct control over the gamerule for this specific death.